lib/model/CResourceMonitor.cc

```diff
@@ -330,12 +338,14 @@ bool CResourceMonitor::isMemoryStable(core_t::TTime bucketLength) const {
     // Coefficient of variation must be less than 0.1
     double mean{maths::CBasicStatistics::mean(m_ModelBytesMoments)};
     double variance{maths::CBasicStatistics::variance(m_ModelBytesMoments)};
-    double cv{(variance > 0.0) ? std::sqrt(variance) / mean : 0.0};
     LOG_TRACE(<< "Model memory stability at " << m_LastMomentsUpdateTime
-              << ": bucket count = " << bucketCount << ", sample count = "
-              << maths::CBasicStatistics::count(m_ModelBytesMoments) << ", mean = " << mean
-              << ", variance = " << variance << ", coefficient of variation = " << cv);
-    if (cv > ESTABLISHED_MEMORY_CV_THRESHOLD) {
+              << ": bucket count = " << bucketCount
+              << ", sample count = " << maths::CBasicStatistics::count(m_ModelBytesMoments)
+              << ", mean = " << mean << ", variance = " << variance
+              << ", coefficient of variation = " << (std::sqrt(variance) / mean));
+    // Instead of literally testing the coefficient of variation it's more
+    // robust against zeroes and NaNs to rearrange it as follows
+    if (variance > maths::CTools::pow2(ESTABLISHED_MEMORY_CV_THRESHOLD * mean)) {
```

Comment:
Regarding your comment on NaNs: is it actually better to return `false` in the case of NaN? Perhaps it is. Unlike 754, C++ doesn't guaranty comparison operator results on NaNs as I recall, so should we replace
```
if (variance > maths::CTools::pow2(ESTABLISHED_MEMORY_CV_THRESHOLD * mean)) {
    return false;
}

return true;
```
with
```
return CMathsFuncs::isNan(variance) &&
       variance <= maths::CTools::pow2(ESTABLISHED_MEMORY_CV_THRESHOLD * mean);
```
Even if we don't explicitly check NaN, it feels slightly cleaner not to bother with the `if`.